### ISSUE TITLE
[nexus,sonarqube] Ingress Resource not compatible with k8s >1.19

### ISSUE THREAD

**Issue by BSchwetzel** [Oteemo/charts#282]
In k8s 1.19 and later the ingress resource created for either nexus or sonarqube is no longer valid. 

This is because of the apiVersion, which should be networking.k8s.io/v1.

Also, the backend definition

```yaml
serviceName: {{ default $serviceName .serviceName }}
servicePort: {{ default $servicePort .servicePort }}
```

should become

```yaml
service:
    name: {{ default $serviceName .serviceName }}
    port: 
        number: {{ default $servicePort .servicePort }}
```

This information can be found in the changelog for 1.19: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md, search for `networking.k8s.io/v1` in this document.

**Comment by rjkernick** [comment]
Good morning, although the new type is available, the old is not fully deprecated until 1.22: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122 . I will update it in the future, but for now it should be okay.

**Comment by betermieux** [comment]
Hi,
1.22 is released and the chart is not installable/upgradable anymore. 
BTW: there are more changes involved, not only changing serviceName and servicePort. Maybe it is easier to create another ingress template which is rendered conditionally.
